mango-core/src/main/scala/org/bdgenomics/mango/models/TrackedLayout.scala

```diff
@@ -57,6 +59,27 @@ object TrackedLayout {
     val ref2 = rec2._1
     ref1.overlaps(ref2)
   }
+
+  def overlaps[T](rec1: (ReferenceRegion, T), recs2: Seq[(ReferenceRegion, T)]): Boolean = {
+    val ref1 = rec1._1
+    val recs2List = recs2.toList
+    val start: Long = recs2List.min(Ordering.by((r: (ReferenceRegion, T)) => r._1.start))._1.start
+    val end: Long = recs2List.max(Ordering.by((r: (ReferenceRegion, T)) => r._1.end))._1.end
+    ref1.overlaps(new ReferenceRegion(recs2.last._1.referenceName, start, end))
+  }
+
+  def overlapsPair[T](rec1: GroupPair, recs2: Seq[(ReferenceRegion, T)]): Boolean = {
+    val ref1 = new ReferenceRegion(recs2.last._1.referenceName, rec1.start, rec1.end)
```

Comment:
I generally prefer `head` to `last`. [`head` is constant time for all Scala collections, while `last` is linear time for some](http://docs.scala-lang.org/overviews/collections/performance-characteristics.html).